Jenkinsfile.dcos-commons

```diff
@@ -15,19 +15,40 @@ pipeline {
  
     stages {
         stage('docker login') {
+            when {
+                anyOf {
+                    branch '*';
+                    // Do not re-build old tags, which were pushed by dockerhub autobuilder
+                    tag pattern "^0\\.5[6-9]\\..*", comparator: "REGEXP";
```

Comment:
What do you think if we change the regex to sth like `^[0-9]{1,2}.[0-9]{1,2}.[0-9]{0,3}$` and do this:
- Try to do a docker pull of the image we want to push.
- If the pull fails, we can continue to push the image.
- If the push succeeds, the image is already present and we SHOULD NOT override the image, this also protects us from overriding an existing image (we can use the jenkins job under `release-tools` if we ever need to override anyway). 

This would also make the regex expr simpler/less redundant. I don't have a hard opinion on this though.

Reply:
> If the pull fails, we can continue to push the image.

I'm afraid that if the pull fails for a reason other than it not existing (like a network flake or timeout) we will misunderstand and potentially overwrite it :-/